File: QytetetJava/src/modeloqytetet/Calle.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package modeloqytetet;

/**
 *
 * @author Juan Carlos Hermoso Quesada
 */
public class Calle extends Casilla{
    
    public Calle(int numeroCasilla, TituloPropiedad titulo) {
        super(numeroCasilla, titulo.getPrecioCompra(), TipoCasilla.CALLE, titulo);
    }
        
     @Override  
    protected TipoCasilla getTipo() {
        return super.getTipo();
    }
    @Override
    protected TituloPropiedad getTitulo() {
        return super.getTitulo();
    }
    
    void asignarPropietarioJugador(Jugador jugador) {
      getTitulo().setPropietario(jugador);
    }
    
    public int pagarAlquiler() {
        return getTitulo().pagarAlquiler();
    }
    @Override
    protected boolean soyEdificable() {
        return super.soyEdificable();
    }
    
    @Override
    public boolean tengoPropietario() {
        return getTitulo().tengoPropietario();
    }
}

File: QytetetRuby/lib/calle.rb
#encoding: utf-8

module ModeloQytetet
  class Calle < Casilla
    def initialize(numeroCasilla, titulo)
      super(numeroCasilla,titulo.precioCompra, TipoCasilla::CALLE, titulo)
    end
    
    def asignarPropietarioJugador(jugador)
      @titulo.propietario = jugador
    end
    
    def pagarAlquiler
      @titulo.pagarAlquiler
    end
    
    def tengoPropietario
      super
    end
  end
end
